src/core/server/logging/logging_config.ts

```diff
@@ -153,7 +167,24 @@ export class LoggingConfig {
       } as AppenderConfigType,
     ],
     [
+      // By default, we transparently rewrite `LogMeta` sent to the
+      // console appender to remove potentially sensitive keys.
+      // This can be overridden in a logger's configuration.
       'console',
+      {
+        type: 'rewrite',
+        appenders: ['stdout'],
+        policy: {
+          type: 'meta',
+          mode: 'update',
+          properties: SENSITIVE_META_PATHS.map((path) => ({ path, value: REDACTED_META_TEXT })),
+        },
+      } as AppenderConfigType,
```

Comment:
> Or filter out sensitive properties when creating meta object

@restrry Yeah, that's the approach we take now, the only issue is that it can't be disabled, and IIRC this was one of the reasons we wanted a rewrite appender to begin with, for the case where we have a reason to allow sensitive headers through for support/debugging purposes.

That's why the only alternative that comes to mind is a new config option to disable filters, which would then be used inside the http response logger `getResponseLog`. But in the spirit of having One Way Of Doing Things (TM), it'd be nice to use the existing config if we can.

Actually, as I think about it, this is an issue that would possibly be solved by [our discussion around having logger-specific default appender configurations](https://github.com/elastic/kibana/issues/90457).

> How it works now in the Legacy platform? Does it always censor the sensitive content?

It filters them out by default (just `authorization` and `cookie`), but you can disable that using the (undocumented) `logging.filter` settings - e.g. `logging.filter.cookie=none` to remove filter from that header in hapi/good. 
https://github.com/elastic/kibana/blob/72670ebc77442a54869f94e7bea2f1f8f3499461/packages/kbn-legacy-logging/src/get_logging_config.ts#L53-L60

Reply:
> Yeah, that's the approach we take now, the only issue is that it can't be disabled, and IIRC this was one of the reasons we wanted a rewrite appender to begin with, for the case where we have a reason to allow sensitive headers through for support/debugging purposes.

I'd say it's nice to have. It is better to not add such a feature than to add it but with the possibility of sensitive data leakage due to incorrect configuration: 
```
 While this ensures nothing sensitive is logged with a default configuration,
the protection is gone as soon as you customize your config to use something other than the console appender. 
```

> Actually, as I think about it, this is an issue that would possibly be solved by our discussion around having logger-specific default appender configurations.

Are you implying that we can hide the Meta object from the logs?  The `pattern` does not apply to `json` layout.

Reply:
> I'd say it's nice to have. It is better to not add such a feature than to add it but with the possibility of sensitive data leakage due to incorrect configuration

Fair point, so do you think we should:
1. Keep the hardcoded filtering of sensitive header fields, meaning we can't turn those off for debugging purposes
2. Go with option 1 but also introduce a new config flag to disable this for debugging purposes (e.g. `logging.filterSensitiveHttpHeaders` or whatever). The flag would default to `true` but could be disabled if someone wanted to use a custom rewrite appender.
3. Some other option I'm not thinking of?

Ideally I'd like to not introduce another config flag, but I'm also unsure of how we could otherwise make it possible to disable the filtering.

> Are you implying that we can hide the Meta object from the logs?

No, sorry that was a bit vague. I meant if we went the route of having predefined default appender configurations on a per-logger basis (similar to the ES config file), then this would be another use case for those and would solve this problem. But if we go the simpler route of removing the `meta` from the `pattern` by default, that won't help us in this case.

Reply:
> Keep the hardcoded filtering of sensitive header fields, meaning we can't turn those off for debugging purposes

As for me, it's an acceptable trade-off. Any other opinions? @joshdover @pgayvallet

Reply:
Yeah the more I think about it, the hardcoded approach feels like the safest option for the time being. And then if we were to consider an enhancement in 8.x to support logger-specific appender defaults as discussed in https://github.com/elastic/kibana/issues/90457, it would allow us to move these settings there anyway.